src/librustc_mir/interpret/step.rs

```diff
@@ -38,6 +38,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     /// Returns `true` as long as there are more things to do.
     ///
     /// This is used by [priroda](https://github.com/oli-obk/priroda)
+    #[inline(always)]
```

Comment:
Please add a comment that this is necessary to make LLVM perform a certain optimization that drastically improves performance when opt-level=3.

Reply:
Wow, inlining such a big function actually *helps*? So weird...

Reply:
https://github.com/rust-lang/rust/pull/67878#issuecomment-570953839

> That might churn the IR enough to give us better performing assembly.

Reply:
Yeah I saw that, but still... weird.

Reply:
It's notable that `step` is only ever used in the tiny function `run` above it in this file, so it isn't going to lead to any huge functions after inlining